sqlglot/tokens.py

```diff
@@ -1187,6 +1187,8 @@ def _scan_number(self) -> None:
             if self._peek.isdigit():
                 self._advance()
             elif self._peek == "." and not decimal:
+                if self.tokens and self.tokens[-1].token_type == TokenType.PARAMETER:
```

Comment:
why would self.tokens be empty?

Reply:
This pattern appears in a couple other places in the tokenizer, it's main purpose afaict is to safeguard against statements which are standalone numbers such as the very first tests in `identity.sql`